vpn-proxy/app/tunnels.py

```diff
@@ -264,14 +263,44 @@ def get_client_conf(tunnel):
 def get_client_script(tunnel):
     return """#!/bin/bash
 
-if ! which openvpn > /dev/null; then
-    if ! which apt-get > /dev/null; then
-        echo "Couldn't find apt-get to install OpenVPN."
+discover_cmd() {
+    echo 'Searching for apt-get, yum, or zypper..'
+    if which apt-get > /dev/null; then
+        pkg_manager='apt-get'
+    elif which yum > /dev/null; then
+        pkg_manager='yum'
+    elif which zypper > /dev/null; then
+        pkg_manager='zypper'
+    else
+        echo "Could not find a package management tool"
```

Comment:
instead of setting variables and using conditionals in many case, just do something like:
```
if which apt-get > /dev/null; then
    apt-get update && apt-get install -yq openvpn
elif which yum > /dev/null; then
   ....
```